tests/test_libs/test_bucket.py

```diff
@@ -0,0 +1,216 @@
+import pytest, mock
+from obs.libs import gmt
+from obs.libs import bucket
+
+
+@pytest.fixture
+def resource(self):
+    pass
+
+
+class Testbucket:
+    def fake_resource(self):
+        resource = mock.Mock()
+        resource.buckets.all.return_value = ["bucket-one", "satu-b21f6bb7-39d5"]
+        resource.Bucket.return_value.objects.filter.return_value = [
+            "bucket-one",
+            "obj.png",
+        ]
+        return resource
+
+    def test_buckets(self):
+        assert bucket.buckets(self.fake_resource()) == [
+            "bucket-one",
+            "satu-b21f6bb7-39d5",
+        ]
+
+    def test_get(self):
+        assert bucket.get_objects(self.fake_resource(), "satu-b21f6bb7-39d5") == [
+            "bucket-one",
+            "obj.png",
+        ]
+
+    def fake_objects(self, resource, bucket_name):
+        obj1 = mock.Mock()
+        obj1.key = "ddg.png"
+        obj1.size = 234
+
+        obj2 = mock.Mock()
+        obj2.key = "obj.txt"
+        obj2.size = 793
+
+        return [obj1, obj2]
+
+    def test_exists(self, monkeypatch):
+        monkeypatch.setattr(bucket, "get_objects", self.fake_objects)
+        assert bucket.is_exists(resource, "satu", "ddg.png") == True
+
+        assert bucket.is_exists(resource, "satu", "ang.png") == False
+
+    def test_bucket_usage(self, monkeypatch):
+        monkeypatch.setattr(bucket, "get_objects", self.fake_objects)
+        assert bucket.bucket_usage(resource, "bucket-name") == (1027, 2)
+
+    def fake_bucket(self, resource):
+        buckets = []
+        for count in range(3):
+            bucket = mock.Mock()
+            bucket.name = "bucket" + str(count)
```

Comment:
since we are using python 3.7+, you can use `f"bucket{count}"` to make it more readable. it's optional though.